### PR TITLE
Allow swapping deployer keys in UserWhitelistContract

### PR DESCRIPTION
**Pull request by mukuls9971**
- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

**Comment by ParthDesai** [comment]
You also need to modify `RemoveUserDeployer` method in deployer whitelist contract, to allow `user-deployer-whitelist` to call it.

**Comment by mukuls9971** [comment]
> Looks alright. Just need to uncomment part blocked by feature.

will do it after txLimiter1285 merging

File: builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go [hunk 1]
```diff
@@ -332,6 +332,88 @@ func (uw *UserDeployerWhitelist) GetDeployedContracts(
...
+
+	// Update deployerState in userDeployerWhitelist contract by adding new state corresponding to NewDeployerAddr with tierID same as oldDeployer and modifying older one corresponding to OldDeployerAddr
+	newDeployer := &UserDeployerState{
+		Address: req.NewDeployerAddr,
+		TierID:  userDeployer.TierID,
+	}
+	if err := ctx.Set(deployerStateKey(loom.UnmarshalAddressPB(req.NewDeployerAddr)), newDeployer); err != nil {
+		return errors.Wrap(err, "Failed to Save WhitelistedDeployer in whitelisted deployers state")
+	}
+	if ctx.FeatureEnabled(loomchain.UserDeployerWhitelistVersion1_2Feature, false) {
```
Comment (enlight) [inline builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go]: This whole method should be gated by this feature flag, so move the feature check up top, right after the basic `req` validation.

File: builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go [hunk 2]
```diff
@@ -332,6 +332,88 @@ func (uw *UserDeployerWhitelist) GetDeployedContracts(
...
+		userDeployer.Inactive = true
+		if err := ctx.Set(deployerStateKey(oldDeployerAddr), &userDeployer); err != nil {
+			return errors.Wrap(err, "Saving WhitelistedDeployer in whitelisted deployers state")
```
Comment (enlight) [inline builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go]: `Saving WhitelistedDeployer in whitelisted deployers state` -> `failed to save old deployer`

File: builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go [hunk 3]
```diff
@@ -332,6 +332,88 @@ func (uw *UserDeployerWhitelist) GetDeployedContracts(
...
+		}
+		return nil
+	}
+
+	ctx.Delete(deployerStateKey(oldDeployerAddr))
```
Comment (enlight) [inline builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go]: This shouldn't ever happen, should only deactivate the deployer.

File: builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go [hunk 4]
```diff
@@ -332,6 +332,88 @@ func (uw *UserDeployerWhitelist) GetDeployedContracts(
...
+			survivedDeployers = append(survivedDeployers, addr)
+		}
+	}
+
+	if !isInputDeployerAddrValid {
+		return ErrDeployerDoesNotExist
+	}
+
+	userState.Deployers = survivedDeployers
+	survivedDeployers = append(survivedDeployers, req.NewDeployerAddr)
```
[the remark this answers is not in the document]
Comment (mukuls9971) [reply builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go]: done

File: Makefile [hunk 5]
```diff
@@ -28,7 +28,7 @@ BINANCE_TGORACLE_DIR=$(GOPATH)/src/$(PKG_BINANCE_TGORACLE)
 # NOTE: To build on Jenkins using a custom go-loom branch update the `deps` target below to checkout
 #       that branch, you only need to update GO_LOOM_GIT_REV if you wish to lock the build to a
 #       specific commit.
-GO_LOOM_GIT_REV = HEAD
+GO_LOOM_GIT_REV = txLimiter1284
```
[the remark this answers is not in the document]
Comment (mukuls9971) [reply Makefile]: done

File: builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go [hunk 6]
```diff
@@ -332,6 +332,90 @@ func (uw *UserDeployerWhitelist) GetDeployedContracts(
...
+	}
+	if err := contract.CallMethod(ctx, dwAddr, "AddUserDeployer", addUserDeployerRequest, nil); err != nil {
+		return errors.Wrap(err, "failed to whitelist deployer")
+	}
+
+	// Update deployerState in userDeployerWhitelist contract by adding new state corresponding to NewDeployerAddr with tierID same as oldDeployer and modifying older one corresponding to OldDeployerAddr
+	newDeployer := &UserDeployerState{
+		Address:   req.NewDeployerAddr,
+		TierID:    userDeployer.TierID,
+		Contracts: userDeployer.Contracts,
```
Comment (enlight) [inline builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go]: I don't recall asking for this to change. The swap is supposed to be equivalent to removing the old key, and adding a new one, without paying the fee again. So the contracts deployed by the old key shouldn't be associated with the new key.

**Pull request merged**